src/diffusers/loaders/unet.py

```diff
@@ -336,8 +340,108 @@ def _process_lora(self, state_dict, unet_identifier_key, network_alphas, adapter
             # otherwise loading LoRA weights will lead to an error
             is_model_cpu_offload, is_sequential_cpu_offload = self._optionally_disable_offloading(_pipeline)
 
-            inject_adapter_in_model(lora_config, self, adapter_name=adapter_name)
-            incompatible_keys = set_peft_model_state_dict(self, state_dict, adapter_name)
+
+            def _check_hotswap_configs_compatible(config0, config1):
+                # To hot-swap two adapters, their configs must be compatible. Otherwise, the results could be false. E.g. if they
+                # use different alpha values, after hot-swapping, the alphas from the first adapter would still be used with the
+                # weights from the 2nd adapter, which would result in incorrect behavior. There is probably a way to swap these
+                # values as well, but that's not implemented yet, and it would trigger a re-compilation if the model is compiled.
+
+                # TODO: This is a very rough check at the moment and there are probably better ways than to error out
+                config_keys_to_check = ["lora_alpha", "use_rslora", "lora_dropout", "alpha_pattern", "use_dora"]
+                config0 = config0.to_dict()
+                config1 = config1.to_dict()
+                for key in config_keys_to_check:
+                    val0 = config0[key]
+                    val1 = config1[key]
+                    if val0 != val1:
+                        raise ValueError(f"Configs are incompatible: for {key}, {val0} != {val1}")
+
+            def _hotswap_adapter_from_state_dict(model, state_dict, adapter_name):
+                """
+                Swap out the LoRA weights from the model with the weights from state_dict.
+
+                It is assumed that the existing adapter and the new adapter are compatible.
+
+                Args:
+                    model: nn.Module
+                        The model with the loaded adapter.
+                    state_dict: dict[str, torch.Tensor]
+                        The state dict of the new adapter, which needs to be compatible (targeting same modules etc.).
+                    adapter_name: Optional[str]
+                        The name of the adapter that should be hot-swapped.
+
+                Raises:
+                    RuntimeError
+                        If the old and the new adapter are not compatible, a RuntimeError is raised.
+                """
+                from operator import attrgetter
```

Comment:
I think `operator` is standard Python. So, okay if it went to the top but I understand why you did it (probably to localize the scope of the changes).

Reply:
Yes, you are correct. Since I know that this function will be moved eventually, I do a local import so that it "moves with the function" :) Once it is finalized, the import can be cleaned up.